src/transaction.rs

```diff
@@ -38,8 +36,11 @@ impl Instruction {
 /// An atomic transaction
 #[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Clone)]
 pub struct Transaction {
-    /// A digital signature of `account_keys`, `program_ids`, `last_id`, `fee` and `instructions`, signed by `Pubkey`.
-    pub signature: Signature,
+    /// A set of digital signature of `account_keys`, `program_ids`, `last_id`, `fee` and `instructions`, signed by `signed_keys`.
+    pub signatures: Vec<Signature>,
+
+    /// The `Pubkeys` that correspond to signature in `signatures`
+    pub signed_keys: Vec<Pubkey>,
```

Comment:
The key duplication between `signed_keys` with `account_keys` is not great.  Seems like there'll be at least 1 duplicated key in almost every transaction.

Reply:
I was thinking the first `signatures.len()` of `account_keys` would be the pubkeys used for signature verification.

Reply:
Well we can do that, when talking with @aeyakovenko I got the impression that we needed a separate `signed_keys` member and that you had proposed that.

Reply:
And that `Transaction::signed_key()` would be changed from just key 0 to any index under `signatures.len()`.

Reply:
I don't have record of it (we should have created an RFC), but I'd guess what was proposed was a separate `signed_keys` (named `signed_account_keys`) and that those same keys should *not* also appear in `account_keys`.  That'd be more intuitive but still a waste of 8 bytes.  I don't see a problem with both signed and unsigned pubkeys being in the same list and the length of the signatures being the trick to know which are which. No bit left behind.

Reply:
@garious for `signed_key` is that an extra index for the signed key then:

```
pub fn signed_key(&self, instruction_index: usize, accounts_index: usize,
signature_index: usize
) -> Option<&Pubkey> {
    match self.key_index(instruction_index, accounts_index) {
        None => None,
        Some(0) => {
               if (signature_index < self.signatures.len() {
                    return self.account_keys.get(0);
               } else {
                    return None;
                }
        Some(_) => None,
    }
}
```

Reply:
yep!